runhouse/rns/folders/folder.py

```diff
@@ -208,15 +224,17 @@ def data_config(self, data_config):
     @property
     def _fs_str(self):
         if isinstance(self.system, Resource):  # if system is a cluster
-            # TODO [DG] Return 'file' if we're on this cluster
+            if self.system.rns_address == _current_cluster("rns_address"):
+                return self.DEFAULT_FS
             return self.CLUSTER_FS
         else:
             return self.system
 
     @property
     def fsspec_fs(self):
-        if self._fsspec_fs is None:
-            self._fsspec_fs = fsspec.filesystem(self._fs_str, **self.data_config)
+        # recompute this each time, since it is different depending if it is being called
+        # from the same cluster (local filesystem) or somewhere else (ssh filesystem)
+        self._fsspec_fs = fsspec.filesystem(self._fs_str, **self.data_config)
```

Comment:
Can you tell if this affects performance? Also, if we're not caching it, we can just return `fsspec.filesystem(self._fs_str, **self.data_config)`, no need to store it in _fsspec_fs I think (unless you see otherwise).

Reply:
I'm not actually too sure the performance impact, and if implemented this way the yea we can get rid of `_fsspec_fs`. Hm not completely sure this is bug proof though, if loading `**self.data_config` from a different file system than it was created on might have issues? (running into a similar issue currently with BERT FT on kitchen branch)